### PR TITLE
Added flag for publishing to the container.

### PR DESCRIPTION
**Pull request by dphulkar-msft**
Added a flag for publishing to the public container.
If the flag is not ticked the artifacts will be released to the test container by default.

Release container name: azcopyvnextrelease
Test container name: azcopyprivatedrops

### CODE REVIEW

File: release-pipeline.yml [hunk 1]
```diff
@@ -7,8 +7,8 @@ parameters:
     type: string
     default: 'azcopy-'
 
-  - name: test_mode
-    displayName: 'Test Mode'
+  - name: publish_to_container
```
Comment (gapra-msft) [inline release-pipeline.yml:10]: Does this flag also prevent actual publishing to the Msft repos or only perform a dry run of publishing to msft repos?

Comment (dphulkar-msft) [reply release-pipeline.yml:10]: for publishing to **https://packages.microsoft.com/** repo we have a different flag named **publish_artifacts** 

**publish_to_container** this flag publishes to **azcopyvnextrelease** container if ticked else it publishes to **azcopyprivatedrops** container which is a test container.
**Test release**: https://ms.portal.azure.com/#view/Microsoft_Azure_Storage/ContainerMenuBlade/~/overview/storageAccountId/%2Fsubscriptions%2Fba45b233-e2ef-4169-8808-49eb0d8eba0d%2FresourceGroups%2FXClient%2Fproviders%2FMicrosoft.Storage%2FstorageAccounts%2Fazcopyprivatedrops/path/releases/etag/%220x8DB86D2AF30CD5B%22/defaultEncryptionScope/%24account-encryption-key/denyEncryptionScopeOverride~/false/defaultId//publicAccessVal/None

File: e2etest/zt_amanaged_disks_test.go [hunk 2]
```diff
@@ -118,7 +119,7 @@ func TestManagedDisks_SnapshotOAuth(t *testing.T) {
 }
 
 // Service issue causes occasional flickers in feature functionality; enough that testing is problematic. Temporarily disabled until issue is resolved.
-func TestManagedDisks_OAuthRequired(t *testing.T) {
+func TestAAA(t *testing.T) {
```
Comment (vibhansa-msft) [inline e2etest/zt_amanaged_disks_test.go:122]: Why is test case name changed to "AAA" ?

**Review by vibhansa-msft — APPROVED** [review]

File: e2etest/zt_amanaged_disks_test.go (hunk 2 above)
Comment (dphulkar-msft) [reply e2etest/zt_amanaged_disks_test.go]: Rolled back the change.

**Review by adreed-msft — APPROVED** [review]

File: e2etest/zt_amanaged_disks_test.go [hunk 3]
```diff
@@ -22,10 +22,11 @@ package e2etest
 
 import (
 	"flag"
-	"github.com/Azure/azure-storage-azcopy/v10/common"
 	"runtime"
 	"testing"
 	"time"
+
```
Comment (vibhansa-msft) [inline e2etest/zt_amanaged_disks_test.go:28]: Can we revert this change as well

Comment (dphulkar-msft) [reply e2etest/zt_amanaged_disks_test.go]: done

**Review by vibhansa-msft — APPROVED** [review]
